### DISCUSSION TITLE
The first possible new feature of C# 13: Params collection

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#89] [General]
# The first possible new feature of C# 13: Params collection

Only one month after the big release of .NET 8, the dotnet team is already working on the next iteration: .NET 9.
With that also comes new language features. The first one is about to be merged into the main development branch: Params Collections.

https://steven-giesel.com/blogPost/5f4fef86-d251-4a47-b893-ca4c515ca314

**Comment by aaronsmithuk** [comment]
Yeah, I'd use it.

**Reply by linkdotnet** [reply]
I didn't use `params` much in my past - but for sure it is nice to have a convenient API, once you really need it.
Probably with some performance benefits!

**Comment by saint4eva** [comment]
Nice writeup. Other features likely to come in C# 13 are readonly parameters in Primary Constructor and support for var in collection expression. As for Descriminated Union, Typed classes and Extensions everything these are really nice to have, but I cannot tell if they will make it to C# 13 - as most of these features would involve runtime support. I think we need to reboot the dotnet Metadata/ IL so we can easily have more modern features such as generics constant e.g. Vector<int, 3>()
